packages/prisma-client/prisma/migrations/20230106000143_tree_styles_data/migration.ts

```diff
@@ -0,0 +1,100 @@
+import { PrismaClient } from "./client";
+
+type StylesItem = {
+  breakpointId: string;
+  instanceId: string;
+  property: string;
+  value: unknown;
+};
+
+type Styles = StylesItem[];
+
+type CssRule = {
+  breakpoint: undefined | string;
+  style: Record<string, unknown>;
+};
+
+type Text = {
+  type: "text";
+  value: string;
+};
+
+type Instance = {
+  type: "instance";
+  id: string;
+  component: string;
+  cssRules: Array<CssRule>;
+  children: Array<Instance | Text>;
+};
+
+const convertTree = (instance: Instance, styles: Styles) => {
+  for (const rule of instance.cssRules) {
+    if (rule.breakpoint === undefined) {
+      continue;
+    }
+    for (const [property, value] of Object.entries(rule.style)) {
+      styles.push({
+        breakpointId: rule.breakpoint,
+        instanceId: instance.id,
+        property,
+        value,
+      });
+    }
+  }
+  instance.cssRules = [];
```

Comment:
Do want to keep `cssRules` property on instance still? Will it always be an empty array now?

Reply:
I think this is here to satisfy typescript and keep potentially not migrated cssRules functional while we can keep running the migration for any amount of time .... basically this is how we would have to do it once we have production sites from customers

Reply:
Yes, I will remove it when normalize tree